src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -166,32 +175,179 @@ public void CompilerDataLogger_Dispose_ShouldReadSarifV1()
+            context.Dispose();
+        }
+
+        [Fact]
+        public void CompilerDataLogger_WriteException_ShouldLogExceptionMessageString()
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            BinaryAnalyzerContext context = CreateTestContext();
+            List<ITelemetry> sendItems = TestSetup(SarifPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            compilerDataLogger.WriteException(context, "testException");
+            sendItems.Count.Should().Be(1);
+            context.Dispose();
+        }
+
+        [Fact]
+        public void CompilerDataLogger_WriteException_ShouldNotLogExceptionMessage_WhenTelemetryClientIsNull()
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            BinaryAnalyzerContext context = CreateTestContext();
+            List<ITelemetry> sendItems = TestSetup(SarifPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object, true);
+            compilerDataLogger.WriteException(context, "testException");
+            sendItems.Count.Should().Be(0);
+            context.Dispose();
+        }
+
+        [Fact]
+        public void CompilerDataLogger_WriteException_ShouldLogExecutionExceptionFromBaseLineTestSarif()
+        {
+            string sarifLogPath = Path.Combine(PEBinaryTests.BaselineTestDataDirectory, ExpectedFolder, SampleSarifPath);
+            var fileSystem = new Mock<IFileSystem>();
+            BinaryAnalyzerContext context = CreateTestContext();
+            List<ITelemetry> sendItems = TestSetup(sarifLogPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            SarifLog sarifLog = SarifLog.Load(sarifLogPath);
+            ExecutionException exception = AnalysisSummaryExtractor.ExtractExceptionData(sarifLog).FirstOrDefault();
+            AnalysisSummary summary = AnalysisSummaryExtractor.ExtractAnalysisSummary(sarifLog, "", null);
+            compilerDataLogger.WriteException(exception, summary);
+            sendItems.Count.Should().Be(1);
+            context.Dispose();
+        }
+
+        [Fact]
+        public void CompilerDataLogger_WriteException_ShouldLogExecutionException()
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            BinaryAnalyzerContext context = CreateTestContext();
+            ExecutionException exception = new ExecutionException(type: "test", message: "testExecutionException", stackTrace: "testStackTrace", innerException: new Exception());
+            AnalysisSummary summary = new AnalysisSummary();
+            List<ITelemetry> sendItems = TestSetup(SarifPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            compilerDataLogger.WriteException(exception, summary);
+            sendItems.Count.Should().Be(1);
```

Comment:
> sendItems.Count.Should().Be(1); [](http://example.com/codeflow?start=11&length=32)

Make sure there is a detail test for validating the specific data logged for an execution exception similar to the summary.